specs/eventing/data-plane.md

```diff
@@ -1,125 +1,184 @@
-# Knative Eventing Data Plane Contracts
+# Knative Eventing Data Plane Contract
 
-## Introduction
-
-Developers using Knative Eventing need to know what is supported for delivery to
-user provided components that receive events. Knative Eventing defines contract
-for data plane components and we have listed them here.
-
-## Conformance
-
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
-"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
-interpreted as described in RFC2119.
-
-## Data plane contract for Sinks
+## Terminology
 
-A **Sink** MUST be able to handle duplicate events.
+This document discusses communication between two parties:
 
-A **Sink** is an [_addressable_](./interfaces.md#addressable) resource that
-takes responsibility for the event. A Sink could be a consumer of events, or
-middleware. A Sink MUST be able to receive CloudEvents over HTTP and HTTPS.
+- **Event Senders** initiate an HTTP POST to deliver a CloudEvent.
+- **Event Recipients** receive an HTTP POST and accept (or reject) a CloudEvent.
 
-A **Sink** MAY be [_callable_](./interfaces.md#callable) resource that
-represents an Addressable endpoint which receives an event as input and
-optionally returns an event to forward downstream.
+Additionally, these roles can be combined in different ways:
 
-Almost every component in Knative Eventing may be a Sink providing
-composability.
+- **Event Processors** can be event senders, event recipients, or both.
+- **Event Sources** are exclusively event senders, and never act as recipients.
+- **Event Sinks** are exclusively event recipients, and never act as senders.
```

Comment:
Does this not completely change the definition of sinks? E.g. we mention channels and brokers as sinks, but they can "send" events to other components. This seems unclear to me and like it might have a bigger impact on docs? What about things like responding to events?

Reply:
I'm not sure we should describe channels and brokers as event sinks. The are Addressable and can receive events, but are more of a step along the way.

Reply:
Perhaps just say Sinks are event recipients, but during their processing they could act in other roles, such as event senders. It gets a bit weird when you think about replies... is that a form of being a "sender" ?